### ISSUE TITLE
Gutenberg time shifts for daylight savings time

### ISSUE THREAD

**Issue by Ipstenu** [WordPress/gutenberg#3271]
## Issue Overview

If you schedule a post for the future AFTER the DST switch, Gutenberg will show the 'correct' time, adjusted for the future, but it won't match the time you're trying to set.

## Steps to Reproduce (for bugs)
NB: DST in US is Nov 5

1. Write a post on Oct 31 
2. Schedule it for 6am on November 9th (AFTER the DST switch in the US)
3. See the date be correct and the time be one hour earlier

![screen shot 2017-10-31 at 31 oct - 12 10 28 pm](https://user-images.githubusercontent.com/807903/32244267-4714ebd6-be35-11e7-918c-0663eafa8940.jpg)

## Expected Behavior
The time should match what you type in (6am) not the time-adjusted time.

## Current Behavior
Time is set an hour 'off'

## Possible Solution
Don't be smart about DST, just accept what I put in 😆

**Comment by noisysocks** [comment]
I'm unable to reproduce this. 

My computer, browser and WordPress installation are set to the Sydney time zone. DST begins (clocks go forward) here on 7 October 2018 at 2am.

I performed these steps:

1. Create a new post
2. Click _Publish..._
3. Expand the _Publish:_ dropdown in the panel that appears
4. Select 7 October 2018, 6am
5. Click _Schedule_ 
6. Close the panel and view the Document inspector 

The inspector displays _October 7, 2018 6:00 am_ as expected.

I also confirmed that the label responds to the date selector as expected and that it doesn't let you input a time that doesn't exist (2am, in this case):

![time](https://user-images.githubusercontent.com/612155/42547641-87c80cd8-8506-11e8-9476-f64b37cf5718.gif)

Going to close this out for now since everything seems to be OK as of Gutenberg 3.2.
